### DISCUSSION TITLE
Theme changes How excactly  is it supposed to  acheived  using the dir of themes?

### DISCUSSION THREAD

**Discussion by plum** [tramhao/termusic#396] [Q&A]
I  would be most grateful If  somebody is able to tell me simply how to switch a tui theme in tui.toml, using the supplied dir  of alternative themes  in ~/.config/termusic/themes. 

WHen I invoke the editor with Shift+c and press tab to select thme aditor,  then press enter, no preview changes occur.

When I save the selected theme, a dialogue pop up informs, 'config change, do I wish to save?,, and I press CTL+S
but no changes are effected. IN a newly automtuc proiduction of a  tui.toml I get the following lines prduced;

```
[theme]
name = "empty name"
author = "empty author"

```

Also, it would be useful to understand what reset means, as for example;

```
[style.important_popup]
foreground_color = "Yellow"
background_color = "Reset"
border_color = "Yellow"
```

Thanks for the great piece of cunningly  useful software.

Cheers, John

**Comment by hasezoey** [comment]
> Also, it would be useful to understand what reset means, as for example;

To my understanding `Reset` is just another "color" in terms of the terminal, ratatui explains it as such:

> Resets the foreground or background color

([ratatui source](https://github.com/ratatui/ratatui/blob/a0979d68715b2c67b32b8909189bc103f0e81f6e/ratatui-core/src/style/color.rs#L69-L71))

> WHen I invoke the editor with Shift+c and press tab to select thme aditor, then press enter, no preview changes occur.

Theoretically, opening the config editor and switching to the `Themes and Colors` tabs and then selecting a Theme and pressing `Enter` on it should preview the colors.
If this does not work, what version of termusic are you running and have your tried clearing the themes directory?
If you did that already, maybe enable debug logs (`RUST_LOG=info,termusic=trace termusic --log-to-file`) and look if there is anything theme related in there.

> IN a newly automtuc proiduction of a tui.toml I get the following lines prduced;

The `empty` fields are because it is the default theme, which does not have its own file (they are hardcoded values) and have no name set. (which i just now realize does not have a way to go back to once having select another theme)

**Reply by plum** [reply]
Thanks for kind answer. I tried again, after clearing the themes directory, with no change in result whith attempts at previewing and switching theme.

The termusic version I am using is 0.9.1-1, obataned from the pacman extras repo.

I shall have to move on now to enabling debug as you suggest.
 Meanwhile I have managed to change to the Belge theme, by manually  copying it into my tui.toml, after deleting the default theme in same conf file. 
That effort  was worth persuing thus far, because I find Belge theme is very nice, much clearer, and with the bonus of appealing aestthetics (to me). I struck lucky there, because I picked on it without any preview!

Cheers, John

**Reply by hasezoey** [reply]
For context, i had used termusic master (d82806cd15a7916bd4a2b1b93434c19c6ea2d1fb) to test the preview, where it "just worked".
In case it matters, what terminal were you using?
